### DISCUSSION TITLE
Marlin 2.1 M112 issued when using ESP3D/ESP8266

### DISCUSSION THREAD

**Discussion by rickjs** [luc-github/ESP3D#948] [Questions]
Marlin 2.1.2.1 on SKR 1.4 board, M112 is issued(some error is causing this) when running, hotend 250c and bed 60c, print idle thru ESP3D/ESP8266 but not when running Octoprint thru USB.

**Comment by luc-github** [comment]
What do you mean by M112 error?
M112 is emmergency stop GCODE command not an error

**Comment by rickjs** [comment]
Either way, ESP3D or ESP8266 is causing it?

**Reply by luc-github** [reply]
Causing what ? as mentionned M112 is not an error, it is a command, so I have asked details

**Reply by rickjs** [reply]
Have printer hotend at 250c and bed 60c controlled with ESP3D/ESP8266, not printing just holding temperature, fans on. After a couple minutes, Marlin issues M112, don't see any errors. As stated, M112: "Used for immediate halt, [M112] shuts down the machine, turns off all the steppers and heaters, and if possible, turns off the power supply. A reset is required to return to operational mode."

**Reply by rickjs** [reply]
As stated running Octoprint using USB is fine except when using websocket thru ESP3D.

**Reply by luc-github** [reply]
M112 is a command that is send by webUI user click on stop button so it is not send by webui autonomously 
There is not websocket support to send command using web socket in ESP3D 2.1, it is only usedto receive serial flow
The issue happen if webUI is open or closed ?
ESP3D itself is not sending any command because it is just a bridge

**Reply by rickjs** [reply]
I'm using socket: on Octoprint to connect to ESP3D, that is not the issue. Isolating the problem, it comes down to ESP3D/ESP8266 issuing or causing an M112 command to Marlin when hotend sits at 250c. Octoprint, Marlin or the SKR board is not doing that.

**Reply by luc-github** [reply]
socket and websocket is not the same 
as mentioned esp3d is just a bridge so M112 does not come from it
can be the lattency using wifi so octoprint think there is some issue and generate M112 for safety

I do not know how you communicate with esp3d using octoprint so I can only imagine also some communication issue

**Reply by rickjs** [reply]
As Sherlock Holmes once said, "If you eliminate the impossible whatever remains, however improbable, must be the truth", what remains is ESP3D or ESP8266.

**Reply by luc-github** [reply]
you forget the way you communicate with esp3d

I suggest you to send The commands to heat using and close immediatly communication with esp3d as well as octoprint
it will tell you if issue comme from esp8266 sending something when nothing is connected to it

**Reply by rickjs** [reply]
Octoprint has a plugin called Netprint which uses socket, ESP3D supports socket. This is not the issue here because ESP3D webui talks directly to the ESP8266 I guess, so this eliminates wifi.

**Reply by luc-github** [reply]
what parameters do you use on  this plugin use to communicate with esp3d by network ?  to understand what is using

**Reply by rickjs** [reply]
Not sure I follow, "I suggest you to send The commands to heat using and close immediatly communication with esp3d as well as octoprint it will tell you if issue comme from esp8266 sending something when nothing is connected to it"  Think I controlled hotend from ESP3D webui directly, even disconnecting Octoprint.

**Reply by luc-github** [reply]
esp3d does not control anything it just forward commands coming from esp3d- webui or your net plugging

**Reply by luc-github** [reply]
so I suggest you to send the heat command and disconnect octo print and webui if open
to see if you reproduce issue

**Comment by rickjs** [comment]
I don't have the printer with me but the socket connection is shown on ESP3D ui, that's pasted in the Octoprint's additonal serial port list. But as I said, I tested not using Octoprint, just using the ESP3D webui, same problem.

**Comment by rickjs** [comment]
I won't be at printer location until Sat. so won't do further testing until then. Thanks

**Comment by luc-github** [comment]
Any update ?

**Reply by rickjs** [reply]
Yes, at first just seemed to be with wifi, then I got it with usb. Changing settings on Octoprint to not cancel was able to get thru most of print. Running into extruded pieces let behind might be creating a problem with the sensorless drivers, sensing collisions. Printing PETG is a problem, readjusting extruder settings in Prusaslicer.

**Reply by luc-github** [reply]
Sorry you wrote before you have issue without octoprint - is that correct ? if that the case you should not search on octprint setting but on EMI noise on your board
As someone wrote 🤓 :
As Sherlock Holmes once said, "If you eliminate the impossible whatever remains, however improbable, must be the truth",

**Reply by luc-github** [reply]
I close issue as not esp3d related

**Reply by rickjs** [reply]
I thought it may have been related to wifi but it happened with usb also
